django_x509/base/models.py

```diff
@@ -429,6 +429,23 @@ def _add_extensions(self, cert):
             ])
         return cert
 
+    def _renew(self):
+        """
+        Renew certificate, private key
+        and validity end date
+        """
+        if hasattr(self, 'ca'):
+            self.ca._generate()
+            self.ca.validity_end = default_ca_validity_end()
+            self.ca.serial_number = uuid.uuid4().int
+            self.ca.save()
+            self.validity_end = default_cert_validity_end()
```

Comment:
wait a minute. If the model has the `ca` attribute it means we're dealing with a certificate.

So if the certificate is being renewed, the renewal of the CA is triggered.
If my understanding of this code is correct, the implementation is wrong.

When renewing a certificate, only the single certificate should be renewed.
When renewing a CA, all the certificates related to it should be renewed.

I would keep this private method, but make sure it only acts on `self`. See below for futher suggestions.